datalad/support/annexrepo.py

```diff
@@ -332,6 +358,144 @@ def annex_add_to_git(self, files):
 
         if self.is_direct_mode():
             cmd_list = ['git', '-c', 'core.bare=false', 'add'] + files
-            self.cmd_call_wrapper.run(cmd_list, cwd=self.path)
+            self.cmd_call_wrapper.run(cmd_list)
+            # TODO: use options with git_add instead!
         else:
             self.git_add(files)
+
+    def annex_initremote(self, name, options):
+        """Creates a new special remote
+
+        Parameters:
+        -----------
+        name: str
+            name of the special remote
+        """
+        # TODO: figure out consistent way for passing options + document
+
+        self._run_annex_command('initremote', annex_options=[name] + options)
+
+    def annex_enableremote(self, name):
+        """Enables use of an existing special remote
+
+        Parameters:
+        -----------
+        name: str
+            name, the special remote was created with
+        """
+
+        self._run_annex_command('enableremote', annex_options=[name])
+
+    @normalize_paths
+    def annex_addurl_to_file(self, file, url, options=[]):
+        """Add file from url to the annex.
+
+        Downloads `file` from `url` and add it to the annex.
+        If annex knows `file` already,
+        records that it can be downloaded from `url`.
+
+        Parameters:
+        -----------
+        file: str
+            technically it's a list, but conversion is done by the decorator
+            and only a single string will work here.
+            TODO: figure out, how to document this behaviour
+                  properly everywhere
+
+        url: str
+
+        options: list
+            options to the annex command
+        """
+
+        annex_options = ['--file=%s' % file[0]] + options + [url]
+        self._run_annex_command('addurl', annex_options=annex_options,
+                                log_online=True, log_stderr=False)
+        # Don't capture stderr, since download progress provided by wget uses
+        # stderr.
+
+    def annex_addurls(self, urls, options=[]):
+        """Downloads each url to its own file, which is added to the annex.
+
+        Parameters:
+        -----------
+        urls: list
+
+        options: list
+            options to the annex command
+        """
+
+        self._run_annex_command('addurl', annex_options=options + urls,
+                                log_online=True, log_stderr=False)
+        # Don't capture stderr, since download progress provided by wget uses
+        # stderr.
+
+    @normalize_paths
+    def annex_rmurl(self, file, url):
+        """Record that the file is no longer available at the url.
+
+        Parameters:
+        -----------
+        file: str
+
+        url: str
```

Comment:
not sure if that empty line would be "proper"... anyways -- eventually we should not allow a commit go through unless docs build fine (see #116)